Kernel.cc

```diff
@@ -1678,23 +1687,23 @@ int Kernel::unlink(char *pathname) {
 			}
 	    }
 	}
+	close(dir);
 
 	// Now we need to free the blocks, if nlinks == 0
 	int nlinks = inode.getNlink()-1;
 	if(nlinks > 0)
 	    inode.setNlink(nlinks);
 	else {
-	    // free any blocks currently allocated to the file
-	    int blockSize = fileSystem->getBlockSize();
-	    int blocks = (inode.getSize() + blockSize-1) / blockSize;
-	    for(int i = 0 ; i < blocks ;i++) {
-		int address = inode.getBlockAddress(i) ;
-		if(address != FileSystem::NOT_A_BLOCK) {
-		    fileSystem->freeBlock(address);
-		    inode.setBlockAddress(i , FileSystem::NOT_A_BLOCK);
-		}
+	    // Search openFiles array for the particular file descriptor (fd was initialized at top)
+	    // to see if any other processes have it open
+	    for(int i = 0; i < MAX_OPEN_FILES; ++i) {
+		if(openFiles[i] == process.openFiles[fd])
```

Comment:
This is comparing a `FileDescriptor*` to a `FileDescriptor**`, so its never gonna work